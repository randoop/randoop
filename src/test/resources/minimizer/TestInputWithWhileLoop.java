import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestInputWithWhileLoop {
  // Test simplification of programs with while loops.
  @Test
  public void test1() throws Throwable {
    int a = 10;
    int x = 0;
    while (x < a) {
      x++;
    }
    org.junit.Assert.assertTrue(x == 0);

    int b = 3;
    while (b > 0) {
      b--;
    }
    org.junit.Assert.assertTrue(b == 0);
  }
}
