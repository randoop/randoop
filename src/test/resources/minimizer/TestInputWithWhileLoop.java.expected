import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestInputWithWhileLoopMinimized {

    // Test simplification of programs with while loops.
    @Test
    public void test1() throws Throwable {
        int a = 10;
        int x = 0;
        while (x < a) {
            x++;
        }
        org.junit.Assert.assertTrue(x == 0);
    }
}
